newIDE/app/src/UI/Confirm/ConfirmDeleteDialog.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+import * as React from 'react';
+import { I18n } from '@lingui/react';
+import { Trans } from '@lingui/macro';
+
+import { type MessageDescriptor } from '../../Utils/i18n/MessageDescriptor.flow';
+import Dialog, { DialogPrimaryButton } from '../Dialog';
+import FlatButton from '../FlatButton';
+import { LargeSpacer } from '../Grid';
+import Text from '../Text';
+import TextField from '../TextField';
+
+type Props = {|
+  open: boolean,
+  title: MessageDescriptor,
+  message: MessageDescriptor,
+  fieldMessage: MessageDescriptor,
+  confirmText: string,
+  onConfirm: () => void,
+  onDismiss: () => void,
+|};
+
+function ConfirmDeleteDialog(props: Props) {
+  const [textInput, setTextInput] = React.useState<string>('');
+  const canConfirm = textInput === props.confirmText;
+
+  const onConfirm = () => {
+    if (!canConfirm) return;
+    props.onConfirm();
```

Comment:
can this error for some reason? should we try catch it to not continue clearing the text input?

Reply:
I read the Provider below, and it looks like the onConfirm does: `setConfirmDeleteDialogOpen(false);` 
So it should be fine and should not error.
However, you're trying to empty the text input after hiding the dialog, so your console is probably unhappy about that?
(I'm not sure you need the line below to clear the input?)

Reply:
The dialog is used in the provider, so at the top level of the app. As a consequence, the component is never unmounted. So at the next dialog appearance, the former text in the input appears.

Reply:
The dialog is used in the provider, so at the top level of the app. As a consequence, the component is never unmounted. So at the next dialog appearance, the former text in the input appears.

Reply:
oh right 👍